### ISSUE TITLE
Accessibility: VoiceOver not announcing accordion aria-expanded state

### ISSUE THREAD

**Issue by brennanyoung** [adaptlearning/adapt_framework#2621]
### Subject of the issue/enhancement/features
With the accordion, VoiceOver does not announce "expanded"/"collapsed" as the aria-expanded attribute is toggled (accordion opened/closed). (NVDA appears to announce it correctly).

This is violation of WCAG success criterion 4.1.2 : "Name, Role, Value", so anybody using the accordion component will be non-compliant with SC 4.1.2

### Your environment
Bug may be reproduced on this adapt demo page
https://adaptlearning.github.io/v4demo/#/id/co-05
VoiceOver on Mac OS 10.15.x

### Steps to reproduce
Start up VoiceOver
Load the URL to the demo page into Safari.
With Keyboard, tab down to focus the first accordion button (labeled "Lists")
Press ENTER to expand it, and again to contract it.

### Expected behaviour
VoiceOver should announce "collapsed" or "expanded" when the accordion is toggled open and closed.

### Actual behaviour
VoiceOver does not announce the change, even though the aria-expanded attribute appears to get toggled correctly. If you tab off the button, then shift-tab back, VoiceOver announces the button's expanded state correctly. It's the toggling itself which does not get announced.

### Further Information:
Earlier versions of VoiceOver ignored aria-expanded, but that has been addressed in more recent versions. This is **not** a VoiceOver bug. [w3c has some disclosure examples](https://w3c.github.io/aria-practices/examples/disclosure/disclosure-navigation.html) which have the correct behavior with VoiceOver. 

Even stranger, the bug disappears if I remove all the adapt javaScript and CSS, and handle the accordion toggle with my own test script.

Here's how I tested that: I copied **all** the markup from inside the body tag of that demo page, and pasted it (unchanged) into the body of a new/empty HTML document. I then added the following bare-bones accordion-toggle script.
```
		<script>
var setupAcc = function (opener, opened) {
	var b = document.querySelector(opener);
	var m = document.querySelector(opened);
	var expand = function () {
			b.setAttribute("aria-expanded", "true");
			m.style.display="block";
	};
	var contract = function () {
			b.setAttribute("aria-expanded", "false");
			m.style.display="none";
	};
	var toggle = function () {
		var expanded = b.getAttribute("aria-expanded") === "true";
		if (expanded) {
			contract();
		} else {
			expand();
		}
	};
	b.onclick = toggle;
}
setupAcc(".accordion-item-title", ".accordion-item-body");
		</script>
```
Once again, tab down to the button labeled "Lists" and press ENTER
With this script, VoiceOver announces "expanded"/"collapsed" correctly. 

Note: The barebones script only hooks up the first of the three accordions, because it uses a class-based lookup of the button and accordion body.

The markup is identical in both cases, but of course there is no CSS or any other scripts from the 'real' adapt page. Very unlikely that CSS is the culprit, so I assume that the adapt javaScript is interfering with VoiceOver somehow. Unfortunately, I have no further clues. 

Good hunting!

**Comment by oliverfoster** [comment]
This is a known issue with voiceover. It will announce the expanded or collapsed state when reentering an expanded or collapsed area but not when toggling.

https://bugs.webkit.org/show_bug.cgi?id=164867

**Comment by brennanyoung** [comment]
webkit issue 164867 is an **iOS** issue.

The w3 example I linked to (and adapt's own accordion demo page markup with my barebones script) do **not** have this issue with **Mac** 10.15.x and VoiceOver. Expanded and collapsed **are** announced when toggling.

**Comment by oliverfoster** [comment]
Try this one:
https://bugs.webkit.org/show_bug.cgi?id=164953

**Comment by brennanyoung** [comment]
> Do you have any references as to when it was fixed?
No, but for the w3c examples, we get no announcement of expanded/collapsed when toggling on 10.14.x whereas we **do** get those announcements on 10.15.x

> Why is it unlikely to be the CSS? Screen readers have CSS related bugs quite regularly.
True, but in my experience the source of Name-Role-Value issues is far more commonly JS than CSS. Therefore I will happily own the prejudice that JS is the more likely culprit, but I have not ruled out CSS involvement entirely. I will not be surprised to discover that Safari is mishandling the attribute somehow under certain circumstances, and needs some special hoop-jumping to do the Right Thing.

> Could you provide all of the files you used for testing?
The only files I used are the w3c disclosure examples (linked above) and the adapt demo page tweaked as described. I've attached my tweaked version of the adapt page here for good measure. Thanks for taking a look!

[disclosure3.zip](https://github.com/adaptlearning/adapt_framework/files/4086386/disclosure3.zip)
